server/src/main/java/org/elasticsearch/transport/Transport.java

```diff
@@ -68,7 +68,10 @@
     /**
      * Returns an address from its string representation.
      */
-    TransportAddress[] addressesFromString(String address, int perAddressLimit) throws UnknownHostException;
+    TransportAddress[] addressesFromString(String address) throws UnknownHostException;
+
+    int[] defaultPortRange();
```

Comment:
I don't think we should add this method to the `Transport` interface just so we can perform the computation in `SettingsBasedSeedHostsProvider`. I think it would work better to have a `getDefaultSeedAddresses` method on `Transport` and `TransportService` and push the computation down into `TcpTransport`.